pkg/stanza/fileconsumer/internal/reader/factory.go

```diff
@@ -76,6 +73,18 @@ func (f *Factory) NewReaderFromMetadata(file *os.File, m *Metadata) (r *Reader,
 		deleteAtEOF:       f.DeleteAtEOF,
 	}
 
+	if r.Fingerprint.Len() > r.fingerprintSize {
+		// User has reconfigured fingerprint_size
+		shorter, rereadErr := fingerprint.NewFromFile(file, r.fingerprintSize)
```

Comment:
Is this a relevant change for this PR?

Reply:
Yeah, it replaces the logic just above, where we were previously able to trim the fingerprint by accessing `FirstBytes` directly. I think it's also in the spirit of the PR because it makes fingerprints effectively immutable, so in order to replace an old one, we must read a new one (although from the same file).

Reply:
Got it, that makes sense to me